clients/drcachesim/drpt2trace/pt2ir.cpp

```diff
@@ -284,30 +289,23 @@ pt2ir_t::convert()
                 return PT2IR_CONV_ERROR_DECODE_NEXT_INSTR;
```

Comment:
We need to do `instrlist_clear_and_destroy` for each of these error cases, right?

Reply:
Done.

Reply:
Drive-by too-late comment, but for future cases: the C++ way would be to create a local object whose destruction will call instrlist_clear_and_destroy() and let the compiler auto-insert the call to it on every return path.

Reply:
Do you mean something like `absl::Cleanup`? Is that an equivalent construct available in DR? Also, we want to _not_ destroy the instrlist in the happy-case.

Reply:
> Do you mean something like absl::Cleanup? Is that an equivalent construct available in DR?

Yes, like absl::Cleanup; no there is no provided class but is simple to make a local struct/class (similar things are done in other places like https://github.com/DynamoRIO/dynamorio/blob/master/clients/drcachesim/tools/opcode_mix.h#L98).

>  Also, we want to not destroy the instrlist in the happy-case.

Oh ok, did not realize this: would take some more logic there so not as clearly a win but can still avoid common errors of forgetting to destroy on some future-added error return.